### ISSUE TITLE
Interest in running in web worker?

### ISSUE THREAD

**Issue by kripken** [playcanvas/engine#11]
I'm experimenting with running webGL apps in web workers, by proxying the GL commands over to the main thread. The motivation is that workers never stall the main thread, can do things the main thread cannot (sync binary xhrs), and also this approach can be faster (10% speedup on emscripten'd cube2/bananabread - queuing up gl commands is faster than actually issuing them, freeing up the cpu-intensive thread).

I started to tinker with playcanvas to see how a non-emscripten app does. I haven't quite gotten this to work, but the gl stream seems to be almost correct. Main issue is I'm not familiar with the playcanvas codebase, so for example as I was tracing back a NaN it went into an onSomething call so I don't have a stack trace, and I got confused.

Any interest in getting playcanvas to run in workers? If so it would be great to work together on this. There's probably not much in the way here - the proxying code passes various emscripten tests, works on cube2/bananabread, and on a playcanvas demo as mentioned above, it emits a gl stream that looks almost right barring some NaNs and wrong # of glDrawElements count. 

Work is at

https://github.com/kripken/webgl-worker

Basically playcanvas/normal.html runs on the main thread, /worker.html in a worker. The last commit adds an option to show just 1 frame in both, for debugging purposes. There's also a debug branch that adds full gl logging on both.
